src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -956,6 +960,14 @@ public bool CleanCacheAndDisposeIfUnused()
                     // Note: Http11 connections will decrement the _associatedHttp11ConnectionCount when disposed.
                     // Http2 connections will not, hence the difference in handing _associatedHttp2ConnectionCount.
                 }
+                if (_availableHttp3Connections is not null)
```

Comment:
Should we keep checking `IsHttp3Supported()` in all the places?

Reply:
`IsHttp3Supported` is guard there for the linker/analyzer complaining that some H/3 code is available on unsupported platforms. Generally if an internal method is called only from a code that's already guarded you don't have to guard again. So I added it only where the compiler complained.